src/hooks/coreSDK/safeCoreSDK.ts

```diff
@@ -44,16 +44,37 @@ export const initSafeSDK = async (
   chainId: string,
   safeAddress: string,
   safeVersion: string,
+  chainInfo: ChainInfo,
+  walletChainId: string,
 ): Promise<Safe> => {
   let isL1SafeMasterCopy = chainId === chains.eth
   // Legacy Safe contracts
   if (isLegacyVersion(safeVersion)) {
     isL1SafeMasterCopy = true
   }
 
-  const ethersProvider = new Web3Provider(provider)
+  const signerProvider = new Web3Provider(provider)
+
+  // Monkey patch the signerProvider to proxy requests to the "readonly" provider if on the wrong chain
+  if (walletChainId !== chainId) {
+    const signerMethods = ['eth_sign', 'eth_signTypedData', 'eth_signTypedData_v4', 'eth_accounts', 'personal_sign']
```

Comment:
this kind of removes support for EIP-712 (eth_signTypedData) signatures. Do we really want that? Also the flow "should" not work with eth_signTypedData (in case it gets there), as by definition in the EIP it should enforce the chainid.

From https://eips.ethereum.org/EIPS/eip-712#definition-of-domainseparator:
> `uint256 chainId` the [EIP-155](https://eips.ethereum.org/EIPS/eip-155) chain id. The user-agent should refuse signing if it does not match the currently active chain.